draft-barnes-acme.md

```diff
@@ -1459,14 +1450,131 @@ TODO
+|   Client   |--------------------+
++------------+                    |
+       ^                          V
+       |   Contact Channel  +------------+
+       +--------------------|    ACME    |
+                            |   Server   |
+                            +------------+
++------------+                    |
+| Validation |<-------------------+
+|   Server   |  Validation Channel
++------------+
+~~~~~~~~~~
+
+In practice, the risks to these channels are not entirely separate, but they are different in most cases.  For example, a CDN employed by the ACME server might be able to tampler with a ACME messages, but not read the client's email (contact channel) or change DNS records for the client's domain names (validation channel).  
+
+Broadly speaking, ACME aims to be secure against active and passive attackers on any individual channel.  Some vulnerabilities arise (noted below), when an attacker can exploit both the ACME channel and one of the others.
+
+On the ACME channel, in addition to network-layer attackers, we also try to defeat an application-layer man in the middle.  Having protections for this case allows an ACME-enabled CA to make use of intermediaries such as Content Distribution Networks (CDNs) with minimal risk.  It also provides defense in depth, e.g., to address cases where a client is behind a middlebox that acts as a man in the middle.
+
+## Integrity of Authorizations
+
+ACME allows anyone to request challenges for an identifier, by registering an account key and sending a new-authorization request under that account key.  The integrity of the authorization process thus depends on the identifier validation challenges to ensure that the challenge can only be completed by someone who both (1) holds the private key of the account key pair, and (2) controls the identifier in question.
+
+All of the challenges above that require an out-of-band query by the server have a binding to the account private key, such that the only the account private key holder can successfully respond to the validation query:
+
+* Simple HTTP: The value provided in the validation request is signed by the account private key.
+* DVSNI: The validation TLS request uses the account key pair as the server's key pair.
+* DNS: The MAC key is derived from an ECDH key for which only the account key holder knows the private key.
+
+The "proof of possession" challenge, which requires no out-of-band query, is bound to an account key by having the prior key sign over the account public key, assuring that the signature cannot be replayed to authorize a different account key.
+
+The association of challenges to identifiers is typically done by requiring the client to respond to a validation query through a channel that only someon in control of the identifier can access.  Thus, each challenge makes an assumption about the underlying channel:
+
+* Simple HTTP: Only the domain owner can provision files under .well-known on a web server for the domain
```

Comment:
Right, and in that case Heroku is violating the assumption that Simple HTTP relies on.